File: src/main/java/cards/Happy.java
package cards;

import com.megacrit.cardcrawl.actions.common.HealAction;
import com.megacrit.cardcrawl.actions.common.MakeTempCardInDiscardAction;
import com.megacrit.cardcrawl.cards.AbstractCard;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import enums.AbstractCardEnum;

public class Happy extends PainCustomCard {

    public static final String ID = "Happy";

    private static final int COST = 1;
    private static final int HEAL_AMT = 3;
    private static final int UPGRADE_HEAL_AMT = 2;

    public Happy() {
        super(ID, COST, CardType.SKILL, AbstractCardEnum.THE_PAIN_PURPLE, CardRarity.SPECIAL, CardTarget.SELF);
        this.baseMagicNumber = this.magicNumber = HEAL_AMT;
        this.exhaust = true;
    }

    @Override
    public void upgrade() {
        if(!upgraded) {
            upgradeName();
            upgradeDescription();
            upgradeMagicNumber(UPGRADE_HEAL_AMT);
        }
    }

    @Override
    public AbstractCard makeCopy() {
        return new Happy();
    }

    @Override
    public void use(AbstractPlayer abstractPlayer, AbstractMonster abstractMonster) {
        AbstractDungeon.actionManager.addToBottom(new HealAction(abstractPlayer, abstractPlayer, this.magicNumber));
        Sad sad = new Sad();
        if(upgraded) sad.upgrade();
        AbstractDungeon.actionManager.addToBottom(new MakeTempCardInDiscardAction(sad, 1));
    }

}

File: src/main/kotlin/kobting/thepain/cards/Happy.kt
package kobting.thepain.cards

import com.megacrit.cardcrawl.actions.common.HealAction
import com.megacrit.cardcrawl.actions.common.MakeTempCardInDiscardAction
import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.dungeons.AbstractDungeon
import com.megacrit.cardcrawl.monsters.AbstractMonster
import kobting.thepain.patches.AbstractCardEnum

class Happy : PainCustomCard(ID, COST, AbstractCard.CardType.SKILL, AbstractCardEnum.THE_PAIN_MAROON, AbstractCard.CardRarity.SPECIAL, AbstractCard.CardTarget.SELF) {
    init {
        this.magicNumber = HEAL_AMT
        this.baseMagicNumber = this.magicNumber
        this.exhaust = true
    }

    override fun upgrade() {
        if (!upgraded) {
            upgradeName()
            upgradeDescription()
            upgradeMagicNumber(UPGRADE_HEAL_AMT)
        }
    }

    override fun makeCopy(): AbstractCard {
        return Happy()
    }

    override fun use(abstractPlayer: AbstractPlayer?, abstractMonster: AbstractMonster?) {
        AbstractDungeon.actionManager.addToBottom(HealAction(abstractPlayer, abstractPlayer, this.magicNumber))
        val sad = Sad()
        if (upgraded) sad.upgrade()
        AbstractDungeon.actionManager.addToBottom(MakeTempCardInDiscardAction(sad, 1))
    }

    companion object {
        @JvmStatic
        val ID = "thepain:Happy"

        private val COST = 1
        private val HEAL_AMT = 3
        private val UPGRADE_HEAL_AMT = 2
    }

}
